File: src/main/scala/com/imcode/imcms/dao/DocLanguageDao.scala
package com.imcode
package imcms.dao

import com.imcode.imcms.mapping.orm.DocLanguage
import org.springframework.transaction.annotation.Transactional
import com.imcode.imcms.dao.hibernate.HibernateSupport

@Transactional(rollbackFor = Array(classOf[Throwable]))
class DocLanguageDao extends HibernateSupport {


  def getAllLanguages(): JList[DocLanguage] = hibernate.listAll()


  def getById(id: Int): DocLanguage = hibernate.get[DocLanguage](id)


  def getByCode(code: String): DocLanguage =
    hibernate.getByNamedQueryAndNamedParams("DocLanguage.getByCode", "code" -> code)


  def saveLanguage(language: DocLanguage): DocLanguage = language |> hibernate.saveOrUpdate


  def deleteLanguage(id: Int) =
    hibernate.bulkUpdateByNamedParams("DELETE FROM DocLanguage l WHERE l.id = :id", "id" -> id)
}

File: src/main/java/com/imcode/imcms/mapping/dao/DocLanguageDao.java
package com.imcode.imcms.mapping.dao;

import com.imcode.imcms.mapping.orm.DocLanguage;
import org.springframework.data.jpa.repository.Modifying;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface DocLanguageDao extends CrudRepository<DocLanguage, Integer> {

    DocLanguage getByCode(String code);

    @Modifying
    @Query("DELETE FROM DocLanguage l WHERE l.code = ?1")
    int deleteByCode(String code);

    List<DocLanguage> findAll();
}
